File: FitBook.Model/Responses/Reservations/ReservationStatusAuditResponse.cs
using FitBook.Model.Enums;

namespace FitBook.Model.Responses.Reservations;

public class ReservationStatusAuditResponse
{
    public int Id { get; set; }
    public ReservationStatus PreviousStatus { get; set; }
    public ReservationStatus NewStatus { get; set; }
    public DateTime ChangedAtUtc { get; set; }
    public string? Reason { get; set; }
    public string ChangedByUserFullName { get; set; } = string.Empty;
}

File: UI/fitbook_desktop/lib/models/responses/reservation_status_audit_response.dart
import 'package:json_annotation/json_annotation.dart';

import '../enums/reservation_status.dart';

part 'reservation_status_audit_response.g.dart';

@JsonSerializable()
class ReservationStatusAuditResponse {
  ReservationStatusAuditResponse({
    required this.id,
    required this.previousStatus,
    required this.newStatus,
    required this.changedAtUtc,
    this.reason,
    required this.changedByUserFullName,
  });

  final int id;
  final ReservationStatus previousStatus;
  final ReservationStatus newStatus;
  final DateTime changedAtUtc;
  final String? reason;
  final String changedByUserFullName;

  factory ReservationStatusAuditResponse.fromJson(Map<String, dynamic> json) =>
      _$ReservationStatusAuditResponseFromJson(json);

  Map<String, dynamic> toJson() => _$ReservationStatusAuditResponseToJson(this);
}
